omod/src/main/java/org/openmrs/module/registrationcore/web/rest/PatientsBySimilarPatientSearchHandler.java

```diff
@@ -0,0 +1,119 @@
+                    .build()));
+    @Override
+    public SearchConfig getSearchConfig() {
+        return this.searchConfig;
+    }
+
+    @Override
+    public PageableResult search(RequestContext requestContext) throws ResponseException {
+        String matchSimilar = requestContext.getParameter(MATCH_SIMILAR);
+
+        if(Boolean.valueOf(matchSimilar)){
+            Patient patient = generatePatientFromParams(requestContext);
+            RegistrationCoreService registrationCoreService = Context.getService(RegistrationCoreService.class);
+            List<Patient> patientList = new LinkedList<Patient>();
+
+
+            List<PatientAndMatchQuality> patientAndMatchQualityList = registrationCoreService.findPreciseSimilarPatients(patient, null, 2d, 10);
+            for(PatientAndMatchQuality matches : patientAndMatchQualityList){
+                patientList.add(matches.getPatient());
+            }
+
+            return new NeedsPaging<Patient>(patientList, requestContext);
+        }
+
+        return new EmptySearchResult();
+    }
+
+    private Patient generatePatientFromParams(RequestContext requestContext) {
+        Person person = new Person();
+        PersonName personName = new PersonName();
+        personName.setGivenName(requestContext.getParameter(GIVENNAME));
+        personName.setMiddleName(requestContext.getParameter(MIDDLENAME));
+        personName.setFamilyName(requestContext.getParameter(FAMILYNAME));
+        person.setNames(Collections.singleton(personName));
+        PersonAddress address = new PersonAddress();
+        address.setAddress1(requestContext.getParameter(ADDRESS_1));
+        address.setAddress2(requestContext.getParameter(ADDRESS_2));
+        address.setCityVillage(requestContext.getParameter(CITY));
+        address.setCountry(requestContext.getParameter(COUNTRY));
+        address.setPostalCode(requestContext.getParameter(POSTALCODE));
+        address.setStateProvince(requestContext.getParameter(STATE));
+        person.setAddresses(Collections.singleton(address));
+        person.setGender(requestContext.getParameter(GENDER));
+        person.setBirthdate(getDateFromString(requestContext.getParameter(BIRTHDATE)));
+
+        return new Patient(person);
+    }
+
+    private Date getDateFromString(String birthdate) {
+        IllegalArgumentException pex = null;
+        String[] supportedFormats = { "yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS",
+                "yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ssXXX", "yyyy-MM-dd'T'HH:mm:ss",
+                "yyyy-MM-dd HH:mm:ss", "yyyy-MM-dd" };
+        for (int i = 0; i < supportedFormats.length; i++) {
+            try {
+                Date date = DateTime.parse(birthdate, DateTimeFormat.forPattern(supportedFormats[i])).toDate();
+                return date;
+            }
+            catch (IllegalArgumentException ex) {
+                pex = ex;
```

Comment:
Why don't you just throw the exception from here and remove the pex variable all together?